docs/architecture/design-proposals/dashboard/deployment-improvements.md

```diff
@@ -0,0 +1,201 @@
+
+## Basic Parameters Definition
+
+The basic parameters that an application supports should be defined per chart. Along with the different files of the application, we should include the information of these parameters as a different file. The proposed approach is to use the file: `values.schema.json`. This file will be a [JSON Schema](https://json-schema.org/). We have chosen this approach since it's also used by Helm v3 to validate the `values.yaml` of a chart ([link](https://github.com/helm/helm/pull/5350/)). Also, it's possible to autogenerate a form from a JSON schema as we do for the service catalog instances. We won't do so for the moment because we cannot be sure that if a `values.schema.json` exists, it represent all the relevants fields of a `values.yaml`. In any case, we won't autogenerate the form to give a better experience.
+
+This JSON Schema allows us to know the structure of the `values.yaml` file but, to know which parameters should be represented in the basic form, we need more information. In particular we need to map well known parameters, like "Disk size", to parameters in the `values.yaml`. For doing so, we can include a new file: `values.form.json`. Having those two pieces of information we are able to deliver the desired experience. See the next section for an example.
+
+### Supported Applications and Parameters for v1
+
+We should decide the first subset of applications and parameters that we want to support in order to deliver the first version of this feature. Based on the popularity of the charts managed by Bitnami that are installed through Kubeapps, we have selected:
+
+1.	bitnami/wordpress
+2.	bitnami/apache
+3.	bitnami/postgresql
+
+Based on those applications, the first batch of parameters we can support is:
+
+- Username. String. Admin username of the application.
+- Password. Password. Admin password of the application.
+- Email. String. Admin email of the application.
+- Enable External Database. Object. Needs to be enabled. Database specifics to be used instead of built-in.
+  - host: String. Database host.
+  - user: String. Database user.
+  - password: Password. Database password.
+  - database: String. Database name.
+  - port: Number. Database port.
+- Enable Hostname. Object. Needs to be enabled. Hostname specifics.
+  - Hostname: String. Hostname to use.
+  - Enable TLS: Boolean. Needs to be enabled. Requires Cert Manager.
+- Disk size. Choice. Sizes to be defined per application. Small, Medium, Large or Other. In case of Other, allows to write a custom disk size.
+- Required Resources. Choice. Resources to be defined per application. Small, Medium, Large or Other. In case of Other, allows to write custom CPU an Memory requirements.
+- Enable Metrics. Boolean. Use Prometheus exporter.
+- Replicas. Number. Number of replicas.
+
+Note that we should be able to represent any parameter in a generic way. We are just defining a set of well-known parameters to be able to render a richer view.
```

Comment:
Again, if we could use the auto-generation based on the schema, for the specific fields, those fields could include an annotation for choices (which turn the field into a select). It seems from the above that you are thinking about standard parameters that we (kubeapps) defines for a richer view, but I think it could remain general (ie. not dependent on kubeapps' current supported parameters for richer views), and instead be based solely on the schema (with extra annotations to identify a field for the form, and choices to present etc.). Does that make sense? Or do you think that would be difficult?